tests/pytests/functional/states/test_mysql.py

```diff
@@ -249,3 +308,89 @@ def test_user_present_absent_unixsocket(mysql, mysql_user, mysql_container):
         if "mariadb" not in mysql_container.mysql_name:
             ret = mysql.plugin_remove("auth_socket")
             assert ret
+
+
+def test_grants_present_absent_state_file(
+    modules, mysql, mysql_grants, mysql_combo, state_tree
+):
+
+    content = """
+    sbclient_2_0:
+      mysql_user.present:
+        - host: localhost
+        - password: sbclient
+        - connection_user: {mysql_user}
+        - connection_pass: {mysql_pass}
+        - connection_db: mysql
+        - connection_port: {mysql_port}
+      mysql_database.present:
+        - connection_user: {mysql_user}
+        - connection_pass: {mysql_pass}
+        - connection_db: mysql
+        - connection_port: {mysql_port}
+      mysql_grants.present:
+        - grant: ALL PRIVILEGES
+        - user: sbclient_2_0
+        - database: sbclient_2_0.*
+        - host: localhost
+        - connection_user: {mysql_user}
+        - connection_pass: {mysql_pass}
+        - connection_db: mysql
+        - connection_port: {mysql_port}
+    """.format(
+        mysql_user=mysql_combo.mysql_root_user,
+        mysql_pass=mysql_combo.mysql_root_passwd,
+        mysql_port=mysql_combo.mysql_port,
+    )
+
+    try:
+        with pytest.helpers.temp_file("manage_mysql.sls", content, state_tree):
+            ret = modules.state.apply(
```

Comment:
If you used `.sls` you get the nicer state return to assert against.
Need to add apply to use that too.